core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1413,19 +1414,31 @@ class ReplicaManager(val config: KafkaConfig,
   def getLogEndOffset(topicPartition: TopicPartition): Option[Long] =
     nonOfflinePartition(topicPartition).flatMap(_.leaderReplicaIfLocal.map(_.logEndOffset))
 
+  private def populateHWMMap(mapping: java.util.HashMap[String, util.HashMap[TopicPartition, Long]], optReplica: Option[Replica]): Unit = {
+    optReplica.foreach(replica => {
+      if (replica.log.isDefined) {
+        val dir = replica.log.get.parentDir
```

Comment:
Cam we rely on `File` instead of converting to `String`? That would avoid a bunch of allocation without having to cache the value.

Reply:
Looking at the code, it seems like `getParentFile` does the allocation too. So, maybe caching the parent dir is needed.